tools/export/sw4stm32/makefile.targets.tmpl

```diff
@@ -1,13 +1,17 @@
 # DO NOT REMOVE! Generated by the SW4STM32 exporter from the mbed project.
 
+ifeq ($(strip $(PREPROC_CMD)),)
+PREPROC_CMD := {{cpp_cmd}}
+endif
```

Comment:
This whole thing (ifeq, assignment, end) is equal to:
```make
PREPROC_CMD ?= {{cpp_cmd}}
```

Reply:
This is even better. "?=" would add value also, if PREPROC_CMD does not have a value.

Reply:
from https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_chapter/make_6.html

There is another assignment operator for variables, `?='. This is called a conditional variable assignment operator, because it only has an effect if the variable is not yet defined. This statement:

FOO ?= bar

is exactly equivalent to this (see section The origin Function):

ifeq ($(origin FOO), undefined)
  FOO = bar
endif